processor/src/aux_table/mod.rs

```diff
@@ -70,15 +66,18 @@ impl AuxTable {
     // PUBLIC ACCESSORS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns the length of the trace required to accommodate co-processor components (excluding
-    /// the padding component).
+    /// Returns the length of the trace required to accommodate co-processor components and 1
+    /// mandatory padding row required for ensuring sufficient trace length for auxiliary connector
+    /// columns that rely on the memory co-processor.
     pub fn trace_len(&self) -> usize {
-        self.hasher.trace_len() + self.bitwise.trace_len() + get_trace_len(&self.memory.trace)
+        self.hasher.trace_len() + self.bitwise.trace_len() + self.memory.trace_len() + 1
     }
 
-    /// Returns all range check lookups required by auxiliary table components.
-    pub fn get_range_checks(&self) -> &RangeCheckMap {
-        &self.memory.range_checks
+    /// Adds all range checks required by the memory co-processor to the provided [RangeChecker]
+    /// instance, along with the cycle rows at which the processor performs the lookups.
+    pub fn update_range_checks(&self, range_checker: &mut RangeChecker) {
+        let memory_start = self.hasher.trace_len() + self.bitwise.trace_len();
+        self.memory.set_range_checks(memory_start, range_checker);
```

Comment:
minor nit: would `append_range_checks` be a better name than `set_range_checks`?